src/dmclock/support/src/indirect_intrusive_heap.h

```diff
@@ -60,7 +60,7 @@ namespace crimson {
 
     static_assert(
       std::is_same<bool,
-      typename std::result_of<C(const T&,const T&)>::type>::value,
```

Comment:
Looks like the new call (std::invoke_result) uses a different syntax. Can you check and fix the same?